README.md

```diff
@@ -11,6 +11,146 @@ This project is an extremely-lightweight task-runner exclusively for __async Pyt
+async def background_task1(state, somearg, somekwarg=True):
+    """`state` must be first argument."""
+    await state.data.get("key")
+    print(state)
+    print(somearg)
+    print(somekwarg)
+
+
+azure_identity_async_credential = DefaultAzureCredential()
+service_bus_namespace_url = os.environ["SERVICE_BUS_NAMESPACE_URL"]
+service_bus_queue_name = os.environ["SERVICE_BUS_QUEUE_NAME"]
+
+# We create a service bus Sender client
+sbus_client = AzureServiceBus(
+    service_bus_namespace_url,
+    service_bus_queue_name,
+    azure_identity_async_credential,  # type: DefaultAzureCredential
+)
+
+# Next we'll create a worker and register our task
+worker = Boilermaker(App({"key": "value"}), service_bus_client=service_bus_client)
+worker.register_async(background_task1, policy=retries.RetryPolicy.default())
+
+
+# Finally, this will schedule the task
+async def publish_task():
+    await worker.apply_async(
+        background_task1, "first-arg", somekwarg=False
+    )
+
+if __name__ == "__main__":
+    asyncio.run(publish_task())
+```
+
+In another process, we'll need to *run* a background task worker. That looks like this:
+
+```python
+async def run_worker():
+    await worker.run()
+```
+
+If we look in the logs for our other process, we should be able to see this task registered and run:
+
+```sh
+{"event": "Registered background function fn=background_task1", "level": "info", "logger": "boilermaker.app", "timestamp": "2024/12/10 15:46:54"}
+{"event": "[background_task1] Begin Task sequence_number=19657", "level": "info", "logger": "boilermaker.app", "timestamp": "2024/12/10 15:52:23"}
+...prints-go-here...
+{"event": "[background_task1] Completed Task sequence_number=19657 in 0.00021130498498678207s", "level": "info", "logger": "boilermaker.app", "timestamp": "2024/12/10 15:52:23"}
+```
+
+
+## Callbacks and Retries
+
+It is possible to register callbacks for tasks, which can run on success or failure. To schedule a task with callbacks, we have to create a task object and then set a success and/or failure callback. Finally, instead of `apply_async` we have to call `publish_task`:
+
+```python
+...see worker-instantation example above...
+
+async def a_background_task(state, param: int):
+    if param > 0:
+        print("Things seem to be going really well")
+        return None
+    raise ValueError("Negative numbers are a bummer")
+
+
+async def happy_path(state):
+    print("Everything is great!")
+
+
+async def sad_path(state):
+    print("Everything is sad")
+
+
+# We need to be sure our tasks are registered
+worker.register_async(a_background_task, policy=retries.NoRetry)
+worker.register_async(happy_path, policy=retries.NoRetry())
+worker.register_async(sad_path, policy=retries.NoRetry())
+
+# Now we can create a happy task and add callbacks
+happy_task = worker.create_task(a_background_task, 11)
+# This callback should get scheduled
+happy_task.on_success = worker.create_task(happy_path)
+# This callback will not
```

Comment:
It took me a second to realize that it won't get scheduled because of the behavior of the example function. Might be worth elaborating in these comments for dummies like me

Reply:
ah, maybe I'll change the params for it: instead of numbers I'll make it strings.

Reply:
How does it look now?

Reply:
Dummy proof, thank you